rali/rali/source/meta_node_rotate.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright (c) 2019 - 2020 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "meta_node_rotate.h"
+void RotateMetaNode::initialize()
+{
+    _src_height_val.resize(_batch_size);
+    _src_width_val.resize(_batch_size);
+    _angle_val.resize(_batch_size);
+}
+float calculate_rotate_point_x(float angle, float x, float cx, float cy)
+{
+    float rotate[2];
+    rotate[0] = cos(RAD(angle));
+    rotate[1] = -1 * sin(RAD(angle));
+    float trans_x = cx + (rotate[0] * cx) + (rotate[1] * cy);
+    return trans_x;
+}
+float calculate_rotate_point_y(float angle, float y, float cx, float cy)
+{
+    float rotate[2];
+    rotate[0] = sin(RAD(angle));
+    rotate[1] = cos(RAD(angle));
+    float trans_y = cy + (rotate[0] * cx) + (rotate[1] * cy);
+    return trans_y;
+}
+void RotateMetaNode::update_parameters(MetaDataBatch* input_meta_data)
+{
+    initialize();
+    if(_batch_size != input_meta_data->size())
+    {
+        _batch_size = input_meta_data->size();
+    }
+    _src_width = _node->get_src_width();
+    _src_height = _node->get_src_height();
+    _dst_width = _node->get_dst_width();
+    _dst_height = _node->get_dst_height();
+    _angle = _node->get_angle();
+    vxCopyArrayRange((vx_array)_src_width, 0, _batch_size, sizeof(uint),_src_width_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+    vxCopyArrayRange((vx_array)_src_height, 0, _batch_size, sizeof(uint),_src_height_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+    vxCopyArrayRange((vx_array)_angle, 0, _batch_size, sizeof(float),_angle_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+    unsigned bb_count;
```

Comment:
Codacy found an issue: [The scope of the variable 'bb_count' can be reduced.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5976200)